packages/core/core/src/BundleGraph.js

```diff
@@ -661,12 +676,9 @@ export default class BundleGraph {
   }
 
   getBundlesInBundleGroup(bundleGroup: BundleGroup): Array<Bundle> {
-    return this._graph
-      .getNodesConnectedFrom(
-        nullthrows(this._graph.getNode(getBundleGroupId(bundleGroup))),
-        'bundle',
-      )
-      .filter(node => node.type === 'bundle')
+    return [...bundleGroup.bundleIds]
+      .reverse()
```

Comment:
The bundler seems to call `addBundleToBundleGroup` in the order `mainbundle, preceding sibling`.  This was determined empirically, not sure if this can be relied upon.